HeterogeneousCore/AlpakaCore/python/ProcessAcceleratorAlpaka.py

```diff
@@ -0,0 +1,70 @@
+import FWCore.ParameterSet.Config as cms
+
+class ModuleTypeResolverAlpaka:
+    def __init__(self, accelerators, backend):
+        # first element is used as the default is nothing is set
+        self._valid_backends = []
+        if "gpu-nvidia" in accelerators:
+            self._valid_backends.append("cuda_async")
+        if "cpu" in accelerators:
+            self._valid_backends.append("serial_sync")
```

Comment:
Do I understand correctly that the choice of backend is made based on the ordering of the (available) backends inside `self._valid_backends` ?
This replaces the "score" used in the `SwitchProducer` ?

Reply:
Correct. As far as the core framework is concerned, the logic is completely up to how `ModuleTypeResolverAlpaka` (in this case) implements it. I went with "list contains all valid backends on the work node with the default as first" just because it felt like a simple way to provide a default backend, and check validity of the backend explicitly set for the module (if set).

In principle the logic can be extended in arbitrary ways.